internal/builder/builder.go

```diff
@@ -373,15 +438,22 @@ func (b *Builder) Save(logger logging.Logger, creatorMetadata CreatorMetadata) e
 		return errors.Wrap(err, "validating buildpacks")
 	}
 
-	if err := validateExtensions(b.lifecycleDescriptor, b.Extensions(), b.additionalExtensions); err != nil {
+	if err := validateExtensions(b.lifecycleDescriptor, b.Extensions(), b.AllModules(buildpack.KindExtension)); err != nil {
 		return errors.Wrap(err, "validating extensions")
 	}
 
 	bpLayers := dist.ModuleLayers{}
 	if _, err := dist.GetLabel(b.image, dist.BuildpackLayersLabel, &bpLayers); err != nil {
 		return errors.Wrapf(err, "getting label %s", dist.BuildpackLayersLabel)
 	}
-	err = b.addModules(buildpack.KindBuildpack, logger, tmpDir, b.image, b.additionalBuildpacks, bpLayers)
+
+	var excludedBuildpacks []buildpack.BuildModule
+	excludedBuildpacks, err = b.addFlattenModules(buildpack.KindBuildpack, logger, tmpDir, b.image, b.additionalBuildpacks.FlattenModules(), bpLayers)
+	if err != nil {
+		return err
+	}
+
+	err = b.addModules(buildpack.KindBuildpack, logger, tmpDir, b.image, append(b.additionalBuildpacks.NonFlattenModules(), excludedBuildpacks...), bpLayers)
```

Comment:
I can't remember, let me check

Reply:
Well, it is actually tested in all the existing tests, because it is returning the Buildpacks as they were before, but I do not think there is a explicit test for it